src/User/Concerns/UserPreferences.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\User\Concerns;
+
+use Illuminate\Support\Arr;
+
+trait UserPreferences
+{
+    /**
+     * An array of registered user preferences. Each preference is defined with
+     * a key, and its value is an array containing the following keys:.
+     *
+     * - transformer: a callback that confines the value of the preference
+     * - default: a default value if the preference isn't set
+     *
+     * @var array
+     */
+    protected static $preferences = [];
```

Comment:
Is there a reason why you don't include disclose_online and follow_after_reply here? If they're the default values...